Liquibase mongodb guide (#20208)
quarkusio/quarkus: docs/src/main/asciidoc/liquibase-mongodb.adoc

The solution is located in the `liquibase-mongodb-quickstart` {quickstarts-tree-url}/liquibase-mongodb-quickstart[directory].

== Setting up support for Liquibase

To start using the Liquibase MongoDB Extension with your project, you just need to:

* add your changeLog to the `{change-log}` file as you usually do with Liquibase
* activate the `migrate-at-start` option to migrate the schema automatically or inject the `Liquibase` object and run
your migration as you normally do.

In your `pom.xml`, add the following dependencies:

* the Liquibase MongoDB extension
* the MongoDB client extension

[source,xml]
----
<dependencies>
    <!-- Liquibase specific dependencies -->
    <dependency>
        <groupId>io.quarkus</groupId>
        <artifactId>quarkus-liquibase-mongodb</artifactId>
    </dependency>

    <!-- MongoDB driver dependencies -->
    <dependency>
        <groupId>io.quarkus</groupId>
        <artifactId>quarkus-mongodb-client</artifactId>
    </dependency>
</dependencies>
----

Liquibase MongoDB extension support relies on the Quarkus MongoDB client config.

> It didn't support multiple MongoDB client for now.

Reviewer: I think we can drop this sentence

> It didn't support multiple MongoDB client for now.

Reviewer: ```suggestion
For the time being, it does not support multiple clients.
```

First, you need to add the MongoDB config to the `{config-file}` file
in order to allow Liquibase to manage the schema.

The following is an example for the `{config-file}` file:

[source,properties]
----
# configure MongoDB
quarkus.mongodb.connection-string = mongodb://localhost:27017

# Liquibase MongoDB minimal config properties
quarkus.liquibase-mongodb.migrate-at-start=true

# Liquibase MongoDB optional config properties
# quarkus.liquibase-mongodb.change-log=db/changeLog.xml
# quarkus.liquibase-mongodb.validate-on-migrate=true
# quarkus.liquibase-mongodb.clean-at-start=false
# quarkus.liquibase-mongodb.contexts=Context1,Context2
# quarkus.liquibase-mongodb.labels=Label1,Label2
# quarkus.liquibase-mongodb.default-catalog-name=DefaultCatalog
# quarkus.liquibase-mongodb.default-schema-name=DefaultSchema
----

Add a changeLog file to the default folder following the Liquibase naming conventions: `{change-log}`

> The yaml, json and xml changeLog file formats are also supported.

Reviewer: ```suggestion
YAML, JSON and XML formats are supported for the changeLog.
```
